avr-hal-generic/src/spi.rs

```diff
@@ -64,220 +66,328 @@ impl Default for Settings {
+    /// back so that its output state can be changed as needed.
+    pub fn new(
+        p: SPI,
+        sclk: port::Pin<port::mode::Output, SCLKPIN>,
+        mosi: port::Pin<port::mode::Output, MOSIPIN>,
+        miso: port::Pin<port::mode::Input<port::mode::PullUp>, MISOPIN>,
+        cs: port::Pin<port::mode::Output, CSPIN>,
+        settings: Settings,
+    ) -> (Self, ChipSelectPin<CSPIN>) {
+        let mut spi = Self {
+            p,
+            sclk,
+            mosi,
+            miso: miso.forget_imode(),
+            write_in_progress: false,
+            _cs: PhantomData,
+            _h: PhantomData,
+        };
+        spi.p.raw_setup(&settings);
+        (spi, ChipSelectPin(cs))
+    }
 
-        impl $Spi<$crate::port::mode::Floating> {
-            /// Instantiate an SPI with the registers, SCLK/MOSI/MISO/CS pins, and settings,
-            /// with an external pull-up on the MISO pin.
-            ///
-            /// The pins are not actually used directly, but they are moved into the struct in
-            /// order to enforce that they are in the correct mode, and cannot be used by anyone
-            /// else while SPI is active.  CS is placed into a `ChipSelectPin` instance and given
-            /// back so that its output state can be changed as needed.
-            pub fn with_external_pullup(
-                peripheral: $SPI,
-                sclk: $sclkmod::$SCLK<$crate::port::mode::Output>,
-                mosi: $mosimod::$MOSI<$crate::port::mode::Output>,
-                miso: $misomod::$MISO<$crate::port::mode::Input<$crate::port::mode::Floating>>,
-                settings: Settings
-            ) -> Self {
-                let spi = $Spi {
-                    peripheral,
-                    sclk,
-                    mosi,
-                    miso,
-                    settings,
-                    is_write_in_progress: false,
-                };
-                spi.setup();
-                spi
+    /// Instantiate an SPI with the registers, SCLK/MOSI/MISO/CS pins, and settings,
+    /// with an external pull-up on the MISO pin.
+    ///
+    /// The pins are not actually used directly, but they are moved into the struct in
+    /// order to enforce that they are in the correct mode, and cannot be used by anyone
+    /// else while SPI is active.
+    pub fn with_external_pullup(
+        p: SPI,
+        sclk: port::Pin<port::mode::Output, SCLKPIN>,
+        mosi: port::Pin<port::mode::Output, MOSIPIN>,
+        miso: port::Pin<port::mode::Input<port::mode::Floating>, MISOPIN>,
+        settings: Settings,
+    ) -> Self {
+        let mut spi = Self {
+            p,
+            sclk,
+            mosi,
+            miso: miso.forget_imode(),
+            write_in_progress: false,
+            _cs: PhantomData,
+            _h: PhantomData,
+        };
+        spi.p.raw_setup(&settings);
+        spi
+    }
```

Comment:
`with_external_pullup()` seems to be missing the chip-select pin?

Reply:
I was copying from here: https://github.com/Rahix/avr-hal/blob/master/avr-hal-generic/src/spi.rs#L198

I'm not too sure why the function on `master` was missing this, but I assumed there was a reason?  If this is just a bug/typo I can add it back in.

Reply:
Most likely something I missed during review of the relevant PR.  It is definitely supposed to be here, otherwise things won't work correctly.